plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,464 @@
+
+		sm := toMap(sentinel)
+		sentinelTags, sentinelFields, err := convertSentinelSentinelsOutput(client.tags, masterName, sm)
+		if err != nil {
+			acc.AddError(err)
+			continue
+		}
+
+		acc.AddFields(measurementSentinels, sentinelFields, sentinelTags)
+	}
+}
+
+// converts `sentinel masters <name>` output to tags and fields
+func convertSentinelMastersOutput(
+	globalTags map[string]string,
+	master map[string]string,
+	quorumErr error,
+) (map[string]string, map[string]interface{}, error) {
+	tags := globalTags
+
+	tags["master"] = master["name"]
+
+	fields, fieldsErr := prepareFieldValues(master, measurementMastersFields)
+
+	if fieldsErr != nil {
+		return nil, nil, fieldsErr
+	}
+
+	fields["has_quorum"] = quorumErr == nil
+
+	return tags, fields, nil
+}
+
+// converts `sentinel sentinels <name>` output to tags and fields
+func convertSentinelSentinelsOutput(
+	globalTags map[string]string,
+	masterName string,
+	sentinelMaster map[string]string,
+) (map[string]string, map[string]interface{}, error) {
+	tags := globalTags
+
+	tags["sentinel_ip"] = sentinelMaster["ip"]
+	tags["sentinel_port"] = sentinelMaster["port"]
+	tags["master"] = masterName
+
+	fields, fieldsErr := prepareFieldValues(sentinelMaster, measurementSentinelsFields)
+
+	if fieldsErr != nil {
+		return nil, nil, fieldsErr
+	}
+
+	return tags, fields, nil
+}
+
+// converts `sentinel replicas <name>` output to tags and fields
+func convertSentinelReplicaOutput(
+	globalTags map[string]string,
+	masterName string,
+	replica map[string]string,
+) (map[string]string, map[string]interface{}, error) {
+	tags := globalTags
+
+	tags["replica_ip"] = replica["ip"]
+	tags["replica_port"] = replica["port"]
+	tags["master"] = masterName
+
+	fields, fieldsErr := prepareFieldValues(replica, measurementReplicasFields)
+
+	if fieldsErr != nil {
+		return nil, nil, fieldsErr
+	}
+
+	return tags, fields, nil
+}
+
+// convertSentinelInfoOutput parses `INFO` command output
+// Largely copied from the Redis input plugin's gatherInfoOutput()
+func convertSentinelInfoOutput(
+	globalTags map[string]string,
+	rdr io.Reader,
+) (map[string]string, map[string]interface{}, error) {
+	scanner := bufio.NewScanner(rdr)
+	rawFields := make(map[string]string)
+
+	tags := globalTags
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 {
+			continue
+		}
+
+		// Redis denotes configuration sections with a hashtag
+		// Example of the section header: # Clients
+		if line[0] == '#' {
+			// Nothing interesting here
+			continue
+		}
+
+		parts := strings.SplitN(line, ":", 2)
+		if len(parts) < 2 {
+			// Not a valid configuration option
+			continue
+		}
+
+		key := strings.TrimSpace(parts[0])
+		val := strings.TrimSpace(parts[1])
+
+		rawFields[key] = val
+	}
+
+	fields, fieldsErr := prepareFieldValues(rawFields, measurementSentinelFields)
+
+	// Rename the field and convert it to nanoseconds
+	fields["uptime_ns"] = int64(time.Duration(fields["uptime_in_seconds"].(int64)) * time.Second)
```

Comment:
Please split this and ensure that the field really contains an `int64` as this will panic otherwise.
```suggestion
	secs, ok := fields["uptime_in_seconds"].(int64)
	if !ok {
		return nil, nil, fmt.Errorf("uptime type %T is not int64", fields["uptime_in_seconds"])
	}
	fields["uptime_ns"] = secs * 1000_000_000
```

Reply:
@srebhan Addressed in https://github.com/influxdata/telegraf/pull/10042/commits/c5f5774106b91300b198fc657e2a47502f614465.